### DISCUSSION TITLE
RuntimeError: Failed to create a default .NET runtime, which would have been "mono" on this system.

### DISCUSSION THREAD

**Discussion by Spanlang** [pythonnet/pythonnet#2013] [Q&A]
Hi,
I have dotnet runtimes and sdks installed on Ubuntu 18.04. 
Python version is 3.8.3 
and 
Pythonnet version is 3.1.0.dev0

dotnet --list-sdks
shows
3.1.404 [/usr/share/dotnet/sdk]
6.0.403 [/usr/share/dotnet/sdk]

dotnet --list-runtimes
shows
Microsoft.AspNetCore.App 3.1.10 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 6.0.11 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.NETCore.App 3.1.10 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
Microsoft.NETCore.App 6.0.11 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

import clr 

results in

RuntimeError: Failed to create a default .NET runtime, which would
                    have been "mono" on this system. Either install a
                    compatible runtime or configure it explicitly via
                    `set_runtime` or the `PYTHONNET_*` environment variables
                    (see set_runtime_from_env).

I tried setting the runtime by hand using 
from pythonnet import load
load("mono")
causing:
RuntimeError: Failed to create a .NET runtime (mono) using the
                parameters {}.

how can I configure the system so that it can find the runtime?

Thanks,
Bernhard

**Comment by filmor** [comment]
All necessary information should be in the documentation: https://pythonnet.github.io/pythonnet/python.html#loading-a-runtime

**Reply by Spanlang** [reply]
Thanks! This is the place where I found out about the setting of the runtime using the load function of pythonnet as I described above.
Do you have any suggestions why that may fail. I could not find the relevant information on that page.

**Reply by filmor** [reply]
The relevant information is that Mono is not .NET Core. Use `"coreclr"` instead and give the .NET Wikipedia page a read, I'll add a few more links to the docs...

**Reply by Spanlang** [reply]
Oh thanks a lot! by switching to coreclr it started to work. I thought I had already tried that but had a typo "corecl" when trying to switch to that.
